clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -672,12 +693,16 @@ public Map<TopicPartition, OffsetAndMetadata> committed(final Set<TopicPartition
     }
 
     private void maybeThrowInvalidGroupIdException() {
-        if (!groupId.isPresent() || groupId.get().isEmpty()) {
-            throw new InvalidGroupIdException("To use the group management or offset commit APIs, you must " +
-                    "provide a valid " + ConsumerConfig.GROUP_ID_CONFIG + " in the consumer configuration.");
+        if (!groupMetadata.isPresent()) {
+            throwInInvalidGroupIdException();
         }
     }
 
+    private void throwInInvalidGroupIdException() {
+        throw new InvalidGroupIdException("To use the group management or offset commit APIs, you must " +
```

Comment:
Kirks' PR gives nicely specific error messages (present, but cannot be empty). I would just inline this. Especially because this is something that people will run into during migration (and it was allowed before), that would be nice to have.